### PR TITLE
feat(atomic): skeleton for Atomic-Rating-Facet

### PR DESCRIPTION
**Pull request by nbuissoncoveo**
https://coveord.atlassian.net/browse/KIT-812

only skeleton for now, afterwards will work on adding the stars.

### CODE REVIEW

File: packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+ * @part values - The facet values container.
+ * @part value-label - The facet value label, common for all displays.
+ * @part value-count - The facet value count, common for all displays.
+ *
+ * @part value-checkbox - The facet value checkbox, available when display is 'checkbox'.
+ * @part value-checkbox-label - The facet value checkbox clickable label, available when display is 'checkbox'.
+ * @part value-link - The facet value when display is 'link'.
+ * @part value-box - The facet value when display is 'box'.
+ *
+ * @part show-more - The show more results button.
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx:49]: Those parts are not available

File: packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+  styleUrl: 'atomic-rating-facet.pcss',
+  shadow: true,
+})
+export class AtomicRatingFacet
+  implements
+    InitializableComponent,
+    BaseFacet<NumericFacet, NumericFacetState> {
+  @InitializeBindings() public bindings!: Bindings;
+  public facet!: NumericFacet;
+  public filter?: NumericFilter;
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx:64]: filter/filterState aren't needed, they are used for the custom input in the other Numeric Facet

File: packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+  @Prop() public field!: string;
+  /**
+   * The number of values to request for this facet, when there are no manual ranges.
+   */
+  @Prop() public numberOfValues = 5;
+  /**
+   * The sort criterion to apply to the returned facet values.
+   * Possible values are 'ascending' and 'descending'.
+   */
+  @Prop() public sortCriteria: RangeFacetSortCriterion = 'ascending';
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx:99]: Might hot needed as well, should always be descending from the designs

File: packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    const options: NumericFacetOptions = {
+      facetId: this.facetId,
+      field: this.field,
+      numberOfValues: this.numberOfValues,
+      sortCriteria: this.sortCriteria,
+      generateAutomaticRanges: false,
+    };
+    this.facet = buildNumericFacet(this.bindings.engine, {options});
+    this.facetId = this.facet.state.facetId;
+    this.bindings.store.state.facets[this.facetId] = {
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx:121]: Once you pull master you'll have to implement the Atomic Store and format similar to the following https://github.com/coveo/ui-kit/pull/952

File: packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+  @State() public isCollapsed = false;
+
+  /**
+   * Specifies a unique identifier for the facet.
+   */
+  @Prop({mutable: true, reflect: true}) public facetId?: string;
+  /**
+   * The non-localized label for the facet.
+   */
+  @Prop() public label = 'rating';
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx:86]: For all facet, we use noLabel as default, to limit the number of strings to be translated in all languages

File: packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+  /**
+   * The field whose values you want to display in the facet.
+   */
+  @Prop() public field!: string;
+  /**
+   * The number of values to request for this facet, when there are no manual ranges.
+   */
+  @Prop() public numberOfValues = 5;
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx:94]: My mistake, updated the document, should be `number-of-stars` like for the rating range facet

File: packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+  /**
+   * The non-localized label for the facet.
+   */
+  @Prop() public label = 'rating';
+  /**
+   * The field whose values you want to display in the facet.
+   */
+  @Prop() public field!: string;
+  /**
+   * The number of values to request for this facet, when there are no manual ranges.
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx:92]: Will have to be updated as there are no manual ranges

**Review by samisayegh — APPROVED** [review]

**Review by ThibodeauJF — APPROVED** [review]

**Pull request merged**